### PR TITLE
feat(Logo / Labels): Consolidation + Props + Modes [GM-149] [GM-146]

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

Integration Branch: https://github.com/codecademy-engineering/Codecademy/pull/24030

* Prunes Logo to 3 Variants (A 4th `proAlt` has been added but not surfaced in the documentation we are given the go ahead to ship it)
* Single Pro Label Version.
* Adds color modes for light and dark versions of label and logo.
* Adds system props `layout` `positioning` and `spacing`.

## Updates
- Curriculum Card now uses the appropriate ProLabel
- GlobalHeader uses the new Logo variants.

### Modes 
Screen Shot 2021-04-26 at 1 31 30 PM

Screen Shot 2021-04-26 at 1 31 40 PM

### PR Checklist

- [x] Related to designs: https://www.figma.com/file/q5fnSuZuxMo5EVS6ZQWzkj/Logos-%26-Labels?node-id=239%3A17
- [x] Related to JIRA ticket: GM-146 GM-149
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.29-alpha.571caa.0<br/>@codecademy/gamut-labs@13.5.6-alpha.571caa.0<br/>@codecademy/gamut@31.0.2-alpha.571caa.0<br/>@codecademy/styleguide@35.0.3-alpha.571caa.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://607cadc0d842d09c2eecb92b--gamut-preview.netlify.app>https://607cadc0d842d09c2eecb92b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/607cadc0d842d09c2eecb92b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.49-alpha.43fa61.0<br/>@codecademy/gamut-labs@14.4.1-alpha.43fa61.0<br/>@codecademy/gamut@32.3.1-alpha.43fa61.0<br/>@codecademy/styleguide@36.4.1-alpha.43fa61.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6086eb4532d34e2ae4cf6214--gamut-preview.netlify.app>https://6086eb4532d34e2ae4cf6214--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6086eb4532d34e2ae4cf6214>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.49-alpha.8ff117.0<br/>@codecademy/gamut-labs@14.4.1-alpha.8ff117.0<br/>@codecademy/gamut@32.3.1-alpha.8ff117.0<br/>@codecademy/styleguide@36.4.1-alpha.8ff117.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6086f599996c643217d6d276--gamut-preview.netlify.app>https://6086f599996c643217d6d276--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6086f599996c643217d6d276>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.49-alpha.514943.0<br/>@codecademy/gamut-labs@14.4.1-alpha.514943.0<br/>@codecademy/gamut@32.3.1-alpha.514943.0<br/>@codecademy/styleguide@36.4.1-alpha.514943.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6086fa937e10874209489816--gamut-preview.netlify.app>https://6086fa937e10874209489816--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6086fa937e10874209489816>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.49-alpha.e4a4a8.0<br/>@codecademy/gamut-labs@14.4.1-alpha.e4a4a8.0<br/>@codecademy/gamut@32.3.1-alpha.e4a4a8.0<br/>@codecademy/styleguide@36.4.1-alpha.e4a4a8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6086febcad07d5008ef67e97--gamut-preview.netlify.app>https://6086febcad07d5008ef67e97--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6086febcad07d5008ef67e97>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.7-alpha.1d53f4.0<br/>@codecademy/gamut-kit@0.3.49-alpha.1d53f4.0<br/>@codecademy/gamut-labs@14.4.1-alpha.1d53f4.0<br/>@codecademy/gamut-styles@9.6.1-alpha.1d53f4.0<br/>@codecademy/gamut-tests@2.3.19-alpha.1d53f4.0<br/>@codecademy/gamut@32.3.1-alpha.1d53f4.0<br/>@codecademy/konamimojisplosion@0.2.19-alpha.1d53f4.0<br/>@codecademy/markdown-overrides@0.5.31-alpha.1d53f4.0<br/>@codecademy/styleguide@36.4.1-alpha.1d53f4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608703e6c658a300bc49a18a--gamut-preview.netlify.app>https://608703e6c658a300bc49a18a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608703e6c658a300bc49a18a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.7-alpha.4382fe.0<br/>@codecademy/gamut-kit@0.3.49-alpha.4382fe.0<br/>@codecademy/gamut-labs@14.4.1-alpha.4382fe.0<br/>@codecademy/gamut-styles@9.6.1-alpha.4382fe.0<br/>@codecademy/gamut-tests@2.3.19-alpha.4382fe.0<br/>@codecademy/gamut@32.3.1-alpha.4382fe.0<br/>@codecademy/konamimojisplosion@0.2.19-alpha.4382fe.0<br/>@codecademy/markdown-overrides@0.5.31-alpha.4382fe.0<br/>@codecademy/styleguide@36.4.1-alpha.4382fe.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60870e8eb685fc0eb3095263--gamut-preview.netlify.app>https://60870e8eb685fc0eb3095263--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60870e8eb685fc0eb3095263>Deploy Logs</a></p>

File: packages/gamut/src/Logo/index.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  pro: { viewBox: '0 0 139 21.705' },
+  mini: { viewBox: '0 0 28 22' },
+};
+
+const accent = {
+  light: 'hyper',
+  dark: 'white',
+} as const;
+
+const Svg = styled('svg', styledConfig)<Omit<LogoProps, 'variant'>>(
```
Comment (dreamwasp) [inline packages/gamut/src/Logo/index.tsx:32]: cool

**Review by dreamwasp — APPROVED** [review]
would like to see a integration branch PR cos i think these old Logos are still used in a couple places, but otherwise ✅

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Something is unhappy with the ProLabel story:

Screenshot of a blank ProLabel

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.7-alpha.a9fd1c.0<br/>@codecademy/gamut-kit@0.3.49-alpha.a9fd1c.0<br/>@codecademy/gamut-labs@14.4.1-alpha.a9fd1c.0<br/>@codecademy/gamut-styles@9.6.1-alpha.a9fd1c.0<br/>@codecademy/gamut-tests@2.3.19-alpha.a9fd1c.0<br/>@codecademy/gamut@32.3.1-alpha.a9fd1c.0<br/>@codecademy/konamimojisplosion@0.2.19-alpha.a9fd1c.0<br/>@codecademy/markdown-overrides@0.5.31-alpha.a9fd1c.0<br/>@codecademy/styleguide@36.4.1-alpha.a9fd1c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60884e40d479690090ab33ee--gamut-preview.netlify.app>https://60884e40d479690090ab33ee--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60884e40d479690090ab33ee>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.53-alpha.41dcc3.0<br/>@codecademy/gamut-labs@14.4.5-alpha.41dcc3.0<br/>@codecademy/gamut@32.4.1-alpha.41dcc3.0<br/>@codecademy/styleguide@36.5.1-alpha.41dcc3.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.53-alpha.fe7bb3.0<br/>@codecademy/gamut-labs@14.4.5-alpha.fe7bb3.0<br/>@codecademy/gamut@32.4.1-alpha.fe7bb3.0<br/>@codecademy/styleguide@36.5.1-alpha.fe7bb3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60885e338ee052008b6b7858--gamut-preview.netlify.app>https://60885e338ee052008b6b7858--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60885e338ee052008b6b7858>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.53-alpha.808e89.0<br/>@codecademy/gamut-labs@14.4.5-alpha.808e89.0<br/>@codecademy/gamut@32.4.1-alpha.808e89.0<br/>@codecademy/styleguide@36.5.1-alpha.808e89.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608863e18dcf6100a2b47bab--gamut-preview.netlify.app>https://608863e18dcf6100a2b47bab--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608863e18dcf6100a2b47bab>Deploy Logs</a></p>

**Comment by codecaaron** [comment]
> Something is unhappy with the ProLabel story:
> 
> Screenshot of a blank ProLabel

Ah whoops this should be fixed!  https://github.com/Codecademy/client-modules/pull/1637/commits/a9fd1cb442aaf951e500dcaedcd36ca5ea67448e#diff-4d2f1d3cc2b180117b7beeedf7d1d5308720bbf2476f9864c0f09a83e96e01b7L13 Old habits :P

**Review by dreamwasp — APPROVED** [review]

File: packages/gamut/src/Logo/index.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  return (
+    <Svg
+      {...props}
+      xmlns="http://www.w3.org/2000/svg"
+      viewBox={viewBox}
+      mode={mode ?? active}
+      height={height}
+    >
+      <title>{title}</title>
+      {variant !== 'mini' && (
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Logo/index.tsx:61]: Nit: maybe use a single ternary, instead of two tenaries with `&&`?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
LGTM overall (yay modes!) but can we separate into logos? I really like the standard we've been setting around having individual components for unique cases...

File: packages/gamut/src/Logo/index.tsx (hunk 2 above)
Comment (codecaaron) [reply packages/gamut/src/Logo/index.tsx:61]: > LGTM overall (yay modes!) but can we separate into logos? I really like the standard we've been setting around having individual components for unique cases...

I tend to agree RE: separate components for static usages like patterns or icons, buuuut I think the the issue is that anywhere they are used we would likely need to reimplement what we lost to internal switching to `pro` `default` and sometimes `mini` in the component where we use them.

Comment (codecaaron) [reply packages/gamut/src/Logo/index.tsx:61]: Do you think that there are enough cases where we won't be implementing this everywhere?

```tsx
import { CodecademyProLogo, CodecademyLogo } from '@codecademy/gamut';

if (isPro) {
  return <CodecademyProLogo />
} else {
 return <CodecademyLogo />
}
```

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.60-alpha.8df6a8.0<br/>@codecademy/gamut-labs@15.1.3-alpha.8df6a8.0<br/>@codecademy/gamut@32.4.4-alpha.8df6a8.0<br/>@codecademy/styleguide@36.6.3-alpha.8df6a8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608ac9c92ac77600935dd67a--gamut-preview.netlify.app>https://608ac9c92ac77600935dd67a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608ac9c92ac77600935dd67a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.60-alpha.a86668.0<br/>@codecademy/gamut-labs@15.1.3-alpha.a86668.0<br/>@codecademy/gamut@32.4.4-alpha.a86668.0<br/>@codecademy/styleguide@36.6.3-alpha.a86668.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608ae37c30093131832a109c--gamut-preview.netlify.app>https://608ae37c30093131832a109c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608ae37c30093131832a109c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.60-alpha.fd47da.0<br/>@codecademy/gamut-labs@15.1.3-alpha.fd47da.0<br/>@codecademy/gamut@32.4.4-alpha.fd47da.0<br/>@codecademy/styleguide@36.6.3-alpha.fd47da.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608ae614b9b24125d8ced92b--gamut-preview.netlify.app>https://608ae614b9b24125d8ced92b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608ae614b9b24125d8ced92b>Deploy Logs</a></p>

**Review by JoshuaKGoldberg — APPROVED** [review]
_perfection_

**Pull request merged**